### PR TITLE
Apply PHP 7.4 syntax and typed property

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since composer.json require php 7.4, php 7.4 syntax can be applied with typed properties. 

- For `final class`, typed properties for protected is allowed if no inherit of parent or used by its final class parent.
- Otherwise, update only private properties.

### CODE REVIEW

**Comment by samsonasik** [comment]
All green 🎉

File: test/CsrfMiddlewareTest.php [hunk 1]
```diff
@@ -15,8 +15,8 @@
 
 class CsrfMiddlewareTest extends TestCase
 {
-    /** @var MockObject<CsrfGuardFactoryInterface> */
-    private $guardFactory;
+    /** @var MockObject&CsrfGuardFactoryInterface */
+    private MockObject $guardFactory;
```
Comment (gsteel) [inline test/CsrfMiddlewareTest.php]: Shouldn't that be:
```php
/** @var MockObject&CsrfGuardFactoryInterface */
private CsrfGuardFactoryInterface $guardFactory;
```

File: test/FlashCsrfGuardTest.php [hunk 2]
```diff
@@ -11,10 +11,9 @@
 
 class FlashCsrfGuardTest extends TestCase
 {
-    /** @var MockObject<FlashMessagesInterface>  */
-    private $flash;
-    /** @var FlashCsrfGuard */
-    private $guard;
+    /** @var MockObject&FlashMessagesInterface  */
+    private MockObject $flash;
```
Comment (gsteel) [inline test/FlashCsrfGuardTest.php]: Same here? More of a question than anything else - I always make the type hint the collaborator/dependency and document the MockObject

File: test/SessionCsrfGuardTest.php [hunk 3]
```diff
@@ -11,10 +11,9 @@
 
 class SessionCsrfGuardTest extends TestCase
 {
-    /** @var MockObject<SessionInterface> */
-    private $session;
-    /** @var SessionCsrfGuard */
-    private $guard;
+    /** @var MockObject&SessionInterface */
+    private MockObject $session;
```
Comment (gsteel) [inline test/SessionCsrfGuardTest.php]: Same question/comment here.

File: test/CsrfMiddlewareTest.php (hunk 1 above)
Comment (samsonasik) [reply test/CsrfMiddlewareTest.php]: updated https://github.com/mezzio/mezzio-csrf/pull/15/commits/049102261d02070f5eed20985ce2e862da9aa811

File: test/FlashCsrfGuardTest.php (hunk 2 above)
Comment (samsonasik) [reply test/FlashCsrfGuardTest.php]: updated https://github.com/mezzio/mezzio-csrf/pull/15/commits/049102261d02070f5eed20985ce2e862da9aa811

File: test/SessionCsrfGuardTest.php (hunk 3 above)
Comment (samsonasik) [reply test/SessionCsrfGuardTest.php]: updated https://github.com/mezzio/mezzio-csrf/pull/15/commits/049102261d02070f5eed20985ce2e862da9aa811

**Comment by samsonasik** [comment]
All green 🎉

**Review by gsteel — APPROVED** [review]
LGTM 👍

**Review by Ocramius — APPROVED** [review]
Thanks @samsonasik!

**Pull request merged**